go-zen-chu/ubuntu-k8s-playbook
# Support automatic k8s upgrade with system-upgrade-controller

## Why

- close: #15
- upgrading k8s each time is time consuming, we need to upgrade automatically

## What

- Since we use k3s, we install system-upgrade-controller to upgrade k8s version to stable

## QA, Evidence

- [x] Checked applying manifest

```bash
$ ansible-playbook -i hosts.yml --limit control_plane1 playbook.yml -t system-upgrade-controller                                                               (git)-[support-automatic-k8s-upgrade] 
SSH password: 
BECOME password[defaults to SSH password]: 

PLAY [inits] ***************************************************************************************************************************************************************************************************************************************************************************
skipping: no hosts matched

PLAY [k8s_control_planes] **************************************************************************************************************************************************************************************************************************************************************

PLAY [k8s_workers] *********************************************************************************************************************************************************************************************************************************************************************
skipping: no hosts matched

PLAY [k8s_control_planes] **************************************************************************************************************************************************************************************************************************************************************

PLAY [k8s_control_planes] **************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

TASK [system-upgrade-controller : Ensure /home/ansible/k8s exists] *********************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

TASK [system-upgrade-controller : Copy manifests] **************************************************************************************************************************************************************************************************************************************
changed: [machine_control_plane1]

TASK [system-upgrade-controller : Apply system-upgrade-controller manifest] ************************************************************************************************************************************************************************************************************
changed: [machine_control_plane1]

TASK [system-upgrade-controller : Copy k8s upgrade plan] *******************************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

TASK [system-upgrade-controller : shell] ***********************************************************************************************************************************************************************************************************************************************
changed: [machine_control_plane1]

PLAY RECAP *****************************************************************************************************************************************************************************************************************************************************************************
machine_control_plane1     : ok=6    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
``` 

- [x] check k8s nodes are upgraded

```bash
$ kgno
NAME    STATUS     ROLES                  AGE    VERSION
node1   Ready      control-plane,master   334d   v1.29.6+k3s2
node2   Ready      <none>                 334d   v1.29.6+k3s2
node3   NotReady   <none>                 334d   v1.27.4+k3s1
```
